File: collaboration_frontend/WebContent/services/userservice.js
/*
 * 
 **/
app.factory('UserService',function($http){
	var userService={}
	var BASE_URL = "http://localhost:8089/CollaborationMiddleware"
	
	userService.registerUser=function(user) {
		
		return $http.post(BASE_URL + "/registeruser",user)
	}
	
	userService.login=function(user){
		return $http.post(BASE_URL + "/login",user)
	}
	
	return userService;
})

File: collaborationbackend/src/main/java/com/hema/collaborationbackend/service/UserService.java
package com.hema.collaborationbackend.service;

import com.hema.collaborationbackend.model.User;

public interface UserService {

	boolean registerUser(User user);
	boolean isUsernameValid(String username);
	boolean isEmailValid(String email);
	User login(User user);
	void update(User user);
	User getUserByUsername(String username);
	 boolean isUpdatedEmailValid(String email,String username);
}
